modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/dnd_table/ContextProvider.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {useCallback, useMemo, useState} from 'react';
+
+import Context from './TableContext';
+
+function ContextProvider({children, columnsNames}) {
+	const [tableWidth, updateTableWidth] = useState(null);
+	const [columnsDefinitions, updateColumnsDefinitions] = useState(new Map());
```

Comment:
In general with plural names you don't want an `s` in the middle; ie:

- `columnsNames` → `columnNames`
- `columnsDefinitions` → `columnDefinitions`